core/src/main/kotlin/plugins/OpenApi.kt

```diff
@@ -119,3 +126,58 @@ fun Application.configureOpenApi() {
         }
     }
 }
+
+/**
+ * The json-schema-serialization library does not properly support nullable types. For example, for a nullable string
+ * property, it creates the following schema:
+ *
+ * ```json
+ * "property": {
+ *   "if": {
+ *     "type": "string"
+ *   },
+ *   "else": {
+ *     "type": "null"
+ *   }
+ * }
+ * ```
+ *
+ * This function replaces this with the correct schema:
+ *
+ * ```json
+ * "property": {
+ *   "type": ["string", "null"]
+ * }
+ * ```
+ */
+private fun JsonElement.replaceIfWithType(): JsonElement {
+    if (this !is JsonObject) {
+        return this
+    }
+
+    val ifElseTypes = getIfElseTypes()
+    val map = toMutableMap()
```

Comment:
Nit: Maybe name this variable `properties` to make it clearer that it contains the data of the current `JsonObject`.